### PR TITLE
perf: improve range gate implementation

### PR DESCRIPTION
**Pull request by mrain**
## Description

closes: #322 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (main)
- [x] Linked to GitHub issue with discussion and accepted design OR have an explanation in the PR that describes this work.
- [x] Wrote unit tests
- [x] Updated relevant documentation in the code
- [x] Added a relevant changelog entry to the `Pending` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the GitHub PR explorer

### CODE REVIEW

**Review by alxiong — COMMENTED** [review]
LGTM! can you add a few unit tests on different bit_length lookups?

File: relation/src/gadgets/ultraplonk/range.rs [hunk 1]
```diff
@@ -28,24 +28,25 @@ impl<F: PrimeField> PlonkCircuit<F> {
         if bit_len == 0 {
```
Comment (ggutoski) [inline relation/src/gadgets/ultraplonk/range.rs:28]: update rustdoc on line 19 above

Comment (mrain) [reply relation/src/gadgets/ultraplonk/range.rs:28]: done in c2c6bdd

**Comment by mrain** [comment]
> LGTM! can you add a few unit tests on different bit_length lookups?

added in c2c6bdd

File: relation/src/gadgets/emulated.rs [hunk 2]
```diff
@@ -318,11 +318,6 @@ impl<F: PrimeField> PlonkCircuit<F> {
             E::NUM_LIMBS,
         );
 
-        // range checking for output c
```
Comment (chancharles92) [inline relation/src/gadgets/emulated.rs:321]: Add a comment on func header that we don't need to rangecheck a and c.

**Review by chancharles92 — COMMENTED** [review]
LGTM

File: relation/src/gadgets/emulated.rs (hunk 2 above)
Comment (mrain) [reply relation/src/gadgets/emulated.rs:321]: added in 9345eea

**Review by chancharles92 — APPROVED** [review]

**Pull request merged**